cardano-testnet/src/Testnet/Start/Cardano.hs

```diff
@@ -405,5 +422,9 @@ cardanoTestnet
       H.assertExpectedSposInLedgerState stakePoolsFp testnetOptions execConfig
 
       pure runtime
+  where
+    writeGenesisSpecFile eraName toWrite = do
```

Comment:
I'm not sure what this will print in the test failure log. Can you check if this should include `HasCallStack` constraint and `withFrozenCallstack`?

Reply:
@carbolymer> Indeed it should both feature `HasCallStack` and `withFrozenCallStack` so changed to:

```hs
    writeGenesisSpecFile :: (MonadTest m, MonadIO m, HasCallStack) => ToJSON a => FilePath -> a -> m ()
    writeGenesisSpecFile eraName toWrite = GHC.withFrozenCallStack $ do
```